web/packages/teleterm/src/services/tshd/grpcContextBridgeClient.ts

```diff
@@ -0,0 +1,198 @@
+  ServerStreamingCall,
+  DuplexStreamingCall,
+  RpcInterceptor,
+  RpcOutputStream,
+  ServiceInfo,
+  RpcError,
+} from '@protobuf-ts/runtime-rpc';
+
+function objectifyRequests<O extends object>(
+  original: RpcInputStream<O>
+): RpcInputStream<O> {
+  return {
+    send: (...args) => original.send(...args),
+    complete: (...args) => original.complete(...args),
+  };
+}
+
+function objectifyResponses<O extends object>(
+  original: RpcOutputStream<O>
+): RpcOutputStream<O> {
+  return {
+    ...original,
+    onMessage: (...args) => original.onMessage(...args),
+    onError: errorCallback =>
+      original.onError(e => errorCallback(objectifyError(e) as Error)),
+    onComplete: (...args) => original.onComplete(...args),
+    onNext: (...args) => original.onNext(...args),
+  };
+}
+
+function objectifyError(error: unknown): object {
+  const e = error as RpcError;
+  return {
+    ...e,
+    message: e.message,
+    cause: e.cause,
+    stack: e.stack,
+    code: e.code,
+    isResolvableWithRelogin: e.meta['is-resolvable-with-relogin'] === '1',
+  };
+}
+
+type PublicProperties<T> = { [K in keyof T]: T[K] }; //keyof only sees public properties
+
+function validateAbortSignalType(
+  abortSignal: AbortSignal | ObjectifiedAbortSignal
+) {
+  if (abortSignal && abortSignal['canBePassedThroughContextBridge'] !== true) {
+    throw new Error(
+      'You must not pass AbortSignal instance. Use objectified version (ObjectifedAbortSignal).'
+    );
+  }
+}
+
+async function objectifyPromiseRejection<TResult>(
+  p: Promise<TResult>
+): Promise<TResult> {
+  try {
+    return await p;
+  } catch (e) {
+    throw objectifyError(e);
+  }
+}
+
+function objectifyThenRejection<TResult>(
+  originalThen: Promise<TResult>['then']
+): Promise<TResult>['then'] {
+  return (onFulfilled, onRejected) => {
+    // If onRejected callback is provided, then it will handle the rejection.
+    if (onRejected) {
+      return originalThen(onFulfilled, reason =>
+        onRejected(objectifyError(reason))
+      );
+    }
+    return objectifyPromiseRejection(originalThen(onFulfilled));
+  };
+}
+
+export function getObjectifiedMethodsInterceptor(): RpcInterceptor {
+  return {
+    interceptUnary(next, method, input, options) {
+      validateAbortSignalType(options.abort);
+
+      const output = next(method, input, options);
+      // UnaryCall is a class with a private field,
+      // so we can't assign an object to it.
+      // The workaround is satisfying only the public part of it
+      // and then casting to UnaryCall.
+      const objectifiedUnaryCall: PublicProperties<UnaryCall> = {
+        method: output.method,
+        request: output.request,
+        requestHeaders: output.requestHeaders,
+        status: objectifyPromiseRejection(output.status),
+        trailers: objectifyPromiseRejection(output.trailers),
+        headers: objectifyPromiseRejection(output.headers),
+        response: objectifyPromiseRejection(output.response),
```

Comment:
I realized that these four promises cause `uncaughtException` errors since we don't handle them.
But I don't know what I can do about it - I believe it would work the same without my interceptors 🤔

Reply:
Where exactly do those errors show up?

Reply:
You can see them, for example, when you switch to the advanced search in the search bar, type 'abc' and press Enter. In the dev tools you will see four unhandled rejections (for `status`, `trailers`, `headers` and `then`). The only promise we are currently handling is `response`.